src/textfield/textfield.js

```diff
@@ -142,10 +142,12 @@
    * @public
    */
   MaterialTextfield.prototype.checkValidity = function() {
-    if (this.input_.validity.valid) {
-      this.element_.classList.remove(this.CssClasses_.IS_INVALID);
-    } else {
-      this.element_.classList.add(this.CssClasses_.IS_INVALID);
+    if (this.input_.validity) {
+      if (this.input_.validity.valid) {
```

Comment:
No worries - I looked at checkDirty first which does that and was thinking the same until I looked closer and saw the logic was reversed...